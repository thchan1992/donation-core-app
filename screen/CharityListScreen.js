import { View, Text, StyleSheet, ScrollView } from "react-native";
import { useEffect } from "react";
import PrimaryButton from "../components/PrimaryButton";
import CharityListItem from "../components/CharityListItem";
import { FlatList } from "react-native-gesture-handler";
import { charitySuggestionScreenMsg } from "../constants/msg";

//This component renders a list of charities on the screen and allows the user to navigate to a specific charity's details screen when a charity is selected from the list. It uses a FlatList to display the data and a CharityListItem component for each charity.
const CharityListScreen = ({ navigation, route }) => {
  // Helper function to render each charity list item
  const renderCharityList = (itemData) => {
    return (
      <CharityListItem
        name={itemData.item.name}
        imageUrl={itemData.item.imageUrl}
        onPress={() => {
          navigation.navigate("CharityScreen", { charity: itemData.item });
        }}
      />
    );
  };
  return (
    <View style={styles.container}>
      <Text style={styles.instrucText}>{charitySuggestionScreenMsg}</Text>
      <FlatList
        data={route.params.charityList}
        keyExtractor={(char) => char.name}
        renderItem={renderCharityList}
        numColumns={1}
      />
    </View>
  );
};
export default CharityListScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  instrucText: {
    alignSelf: "center",
    padding: 5,
    fontWeight: "bold",
    fontSize: 25,
    color: "white",
  },
});
