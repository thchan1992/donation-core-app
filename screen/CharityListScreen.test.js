import React from "react";
import renderer from "react-test-renderer";
import CharityListScreen from "./CharityListScreen";

describe("CharityListScreen", () => {
  const navigation = { navigate: jest.fn() };
  const route = {
    params: { charityList: [{ name: "charity 1", imageUrl: "image.com" }] },
  };
  let component;
  beforeEach(() => {
    component = renderer.create(
      <CharityListScreen navigation={navigation} route={route} />
    );
  });

  it("should match the snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should call navigation.navigate when item is pressed", () => {
    const instance = component.root;
    const charityListItem = instance.find(
      (el) => el.type.name === "CharityListItem"
    );
    charityListItem.props.onPress();
    expect(navigation.navigate).toHaveBeenCalledWith("CharityScreen", {
      charity: expect.any(Object),
    });
  });
});
