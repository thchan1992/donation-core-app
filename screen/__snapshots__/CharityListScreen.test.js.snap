// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityListScreen should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "padding": 5,
      }
    }
  >
    Here are the suggestions. Please choose any charities below.
  </Text>
  <RCTScrollView
    collapsable={false}
    data={
      Array [
        Object {
          "imageUrl": "image.com",
          "name": "charity 1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#28C7FA",
              "borderColor": "#FF304F",
              "borderRadius": 8,
              "borderWidth": 1.5,
              "height": 180,
              "margin": 10,
              "opacity": 1,
              "width": 150,
            }
          }
        >
          <Image
            source="image.com"
            style={
              Object {
                "alignSelf": "center",
                "borderColor": "#FF304F",
                "borderRadius": 8,
                "borderWidth": 1.5,
                "height": 150,
                "justifyContent": "center",
                "width": 150,
              }
            }
          />
          <View>
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "black",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "padding": 3,
                    "textTransform": "uppercase",
                  },
                  Object {
                    "alignSelf": "center",
                    "color": "black",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "padding": 3,
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              charity 1
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
